test/software/amazon/ion/EquivsTestCase.java

```diff
@@ -187,12 +191,45 @@ protected void checkEquivalence(final IonValue left,
         }
     }
 
+    public static IonDatagram[] roundTripDatagram(IonDatagram input) throws IOException{
+        IonSystem system = IonSystemBuilder.standard().build();
+        IonLoader loader = system.getLoader();
+        ByteArrayOutputStream textOutputStream = new ByteArrayOutputStream();
+        ByteArrayOutputStream binaryOutputStream = new ByteArrayOutputStream();
+        IonWriter textWriter = IonTextWriterBuilder.standard().build(textOutputStream);
```

Comment:
Initialization of the writers needs to take place inside the `try`. Declare outside, initialize inside. Same with the reader. Also, the reader should be closed in the `finally` too.